libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -194,24 +193,74 @@ impl<'a> LiftAstToDml<'a> {
     ) -> Result<(dml::FieldType, Vec<ast::Directive>), DatamodelError> {
         let type_name = &ast_field.field_type.name;
 
+        let mut supports_native_types = false;
+        let mut datasource_name = "";
```

Comment:
You don't need mutability here. A `match` is an expression you can use the result. I'd write it like this:
```
let (supports_native_types, datasource_name) = match self.source {
  Some(source) => (source.has_preview_feature("native_types"), source.name.as_str()),
  ...
}
```